rust/lance/src/index/vector/ivf/builder.rs

```diff
@@ -89,14 +98,109 @@ pub(super) async fn build_partitions(
     Ok(())
 }
 
+async fn load_precomputed_partitions(
+    src_dataset: &Dataset,
+    partitions_ds_uri: &str,
+) -> Result<Vec<Vec<i32>>> {
```

Comment:
Use `Vec<Vec<u32>>` to avoid casting?

Reply:
I'm using `i32` so that I can use -1 as a placeholder for deleted rows (there is no guarantee that every address still exists in the dataset).